gateway/sds_gateway/visualizations/views.py

```diff
@@ -25,3 +27,30 @@ def get_context_data(self, **kwargs):
 
         context["capture"] = capture
         return context
+
+
+@method_decorator(login_required, name="dispatch")
+class SpectrogramVisualizationView(LoginRequiredMixin, TemplateView):
+    """View for displaying spectrogram visualization of a capture."""
+
+    template_name = "visualizations/spectrogram.html"
+
+    def dispatch(self, request, *args, **kwargs):
+        """Check if spectrogram feature is enabled before processing the request."""
+        if not settings.EXPERIMENTAL_SPECTROGRAM:
+            error_message = "Spectrogram feature is not enabled"
+            raise Http404(error_message)
+        return super().dispatch(request, *args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        """Get context data for the spectrogram visualization."""
+        context = super().get_context_data(**kwargs)
+
+        # Get the capture from the URL parameter
+        capture_uuid = self.kwargs.get("capture_uuid")
+        capture = get_object_or_404(
+            Capture, uuid=capture_uuid, owner=self.request.user, is_deleted=False
+        )
+
```

Comment:
FYI, this is limiting visualizations to owned captures, which is probably fine initially (especially to not extend this PR even more), but I think eventually we'll want to expand this to shared captures too.